test/e2e_node/device_plugin_test.go

```diff
@@ -264,9 +264,13 @@ func testDevicePlugin(f *framework.Framework, pluginSockDir string) {
 
 			ginkgo.By("Recreating the plugin pod")
 			devicePluginPod = e2epod.NewPodClient(f).CreateSync(ctx, dptemplate)
+			err = e2epod.WaitTimeoutForPodRunningInNamespace(ctx, f.ClientSet, devicePluginPod.Name, devicePluginPod.Namespace, 1*time.Minute)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Confirming that after a kubelet restart, fake-device assignment is kept")
+			err = e2epod.WaitTimeoutForPodRunningInNamespace(ctx, f.ClientSet, pod1.Name, f.Namespace.Name, 1*time.Minute)
```

Comment:
I see what you mean but my intention here was to capture the fact that after kubelet restarted and the device plugin re-registered we confirm the device assignment. Admittedly, the statement in the `By` clause does not reflect that accurately. The check is not immediately after kubelet restart but it is still after kubelet restart :) Perhaps it would be better to update the statement or move the check right after kubelet restart.
I will make update to address this.